src/score_components/find-problematic-comments.js

```diff
@@ -31,13 +31,22 @@ module.exports = function grepForProblematicComments(ext, include, exclude) {
       find += ` -not -path "*/${ex}"`;
     }
   });
-  find += ' -exec grep -E \'TODO|HACK|FIXME\' {} \\;';
-  let output;
+  const commentPattern = '\\s*(//|/\\*|\\*).*\\b(TODO|HACK|FIXME)\\b';
+
+  // Modify the grep command to search within comments only
+  find += ` -exec sh -c 'grep -EHn "${commentPattern}" "$0"' {} \\;`;
+  let output = '';
   try {
     output = child_process.execSync(find).toString().trim();
   } catch (e) {
-    // TODO: handle error
-    output = ''; // temporary fix to avoid undefined
+    core.error('child_process execSync failed to execute');
```

Comment:
Same comment here re: pass `e` to `error`

Reply:
Done!